presto-client/src/main/java/com/facebook/presto/client/StatementClientFactory.java

```diff
@@ -15,12 +15,29 @@
 
 import okhttp3.OkHttpClient;
 
+import static java.util.Objects.requireNonNull;
+
 public final class StatementClientFactory
 {
+    public static final ProtocolVersion DEFAULT_PROTOCOL_VERSION = ProtocolVersion.V1;
```

Comment:
I would remove this static field, instead I would inject here protocol version to factory construct from configuration/cli parameters and create an instance of factory that would be used multiple times. What do you think?

Reply:
Both v1 and v2 might be used by different statement clients in the same process, for example unit tests use it. As a result, it cannot be a singleton initialized at start up. What class/component will then own an instance of the version-specific factory?